rsmtool/rsmexplain.py

```diff
@@ -0,0 +1,525 @@
+        )
+
+    # find the rsmtool experiment directory
+    experiment_dir = DataReader.locate_files(
+        configuration["experiment_dir"], configuration.configdir
+    )
+    if not experiment_dir:
+        raise FileNotFoundError(f"The directory {configuration['experiment_dir']} does not exist.")
+    else:
+        experiment_output_dir = normpath(join(experiment_dir, "output"))
+        if not exists(experiment_output_dir):
+            raise FileNotFoundError(
+                f"The directory {experiment_dir} does not contain "
+                f"the output of an rsmtool experiment."
+            )
+
+    # find all the .model files in the experiment output directory
+    model_files = glob.glob(join(experiment_output_dir, "*.model"))
+    if not model_files:
+        raise FileNotFoundError(
+            f"The directory {experiment_output_dir} does not contain any rsmtool models."
+        )
+
+    experiment_ids = [splitext(basename(mf))[0] for mf in model_files]
+    if experiment_id not in experiment_ids:
+        raise FileNotFoundError(
+            f"{experiment_output_dir} does not contain a model "
+            f'for the experiment "{experiment_id}". The following '
+            f"experiments are contained in this directory: {experiment_ids}"
+        )
+
+    # check that the directory contains outher required files
+    expected_feature_file_name = f"{experiment_id}_feature.csv"
+    if not exists(join(experiment_output_dir, expected_feature_file_name)):
+        raise FileNotFoundError(
+            f"{experiment_output_dir} does not contain the "
+            f"required file {expected_feature_file_name} that was "
+            f"generated during model training."
+        )
+
+    # load the background and explainable data sets
+    (background_data_path, explainable_data_path) = DataReader.locate_files(
+        [configuration["background_data"], configuration["explainable_data"]],
+        configuration.configdir,
+    )
+    if not background_data_path:
+        raise FileNotFoundError(f"Input file {configuration['background_data']} does not exist")
+    if not explainable_data_path:
+        raise FileNotFoundError(f"Input file {configuration['explainable_data']} does not exist")
+
+    # read the background data, explainable data, and feature info files
+    feature_info = join(experiment_output_dir, f"{experiment_id}_feature.csv")
+    file_paths = [background_data_path, explainable_data_path, feature_info]
+    file_names = [
+        "background_features",
+        "explainable_features",
+        "feature_info",
+    ]
+    reader = DataReader(file_paths, file_names)
+    container = reader.read(kwargs_dict={"feature_info": {"index_col": 0}})
+
+    # ensure that the background data is large enough for meaningful explanations
+    background_data_size = len(container.background_features)
+    try:
+        assert background_data_size > 300
+    except AssertionError:
```

Comment:
1. Wouldn't it be better to use a simple if and raise an error?
2. currently it just exits, without raising the error (as stated in the feature docstring).
3. should it be >= 300?

Reply:
All good points. Will fix!